backend/src/test/java/de/neuefische/CapStone/backend/controller/HubControllerTest.java

```diff
@@ -0,0 +1,144 @@
+package de.neuefische.CapStone.backend.controller;
+
+import de.neuefische.CapStone.backend.api.Hub;
+import de.neuefische.CapStone.backend.config.JwtConfig;
+import de.neuefische.CapStone.backend.model.HubEntity;
+import de.neuefische.CapStone.backend.model.UserEntity;
+import de.neuefische.CapStone.backend.repo.HubRepository;
+import de.neuefische.CapStone.backend.repo.UserRepository;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+@SpringBootTest(
+        properties = "spring.profiles.active:h2",
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+public class HubControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    public HubControllerTest(HubRepository hubRepository, UserRepository userRepository, PasswordEncoder passwordEncoder, TestRestTemplate testRestTemplate, JwtConfig jwtConfig) {
+        this.hubRepository = hubRepository;
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.testRestTemplate = testRestTemplate;
+        this.jwtConfig = jwtConfig;
+    }
+
+
+    private String url() {
+        return "http://localhost:" + port;
+    }
+    private final HubRepository hubRepository;
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final TestRestTemplate testRestTemplate;
+    private final JwtConfig jwtConfig;
+
+    @AfterEach
+    public void clearUserRepo() {
+        hubRepository.deleteAll();
+        userRepository.deleteAll();
+    }
+
+    @BeforeEach
+    public void setUser() {
+        userRepository.save(UserEntity.builder()
```

Comment:
saveAndFlush()   ?

Reply:
ist eine eigene Methode, kann man machen, aber solange das in einer eigenen Methode ist wirds jedes mal beim rausgehen gesaved. SaveAndFlush macht nur Sinn wenn man in der gleichen Methode auf die Datenbank zugreift (so habe ichs zumindest verstanden)